js/deribit.js

```diff
@@ -1790,16 +1793,29 @@ module.exports = class deribit extends Exchange {
         for (let i = 0; i < positions.length; i++) {
             result.push (this.parsePosition (positions[i]));
         }
-        // todo unify parsePositions
         return result;
     }
 
     async fetchPositions (symbols = undefined, params = {}) {
         await this.loadMarkets ();
-        const code = this.codeFromOptions ('fetchPositions', params);
+        let code = undefined;
+        if (symbols === undefined) {
+            code = this.codeFromOptions ('fetchPositions', params);
+        } else if (typeof symbols === 'string') {
+            code = symbols;
+        } else {
+            if (Array.isArray (symbols)) {
+                const length = symbols.length;
+                if (length !== 1) {
+                    throw new BadRequest (this.id + ' fetchPositions symbols argument cannot contain more than 1 symbol');
+                }
+                code = symbols[0];
+            }
+        }
```

Comment:
We are mixing types here, the `symbols`, if defined, is an array of market symbols.
The `code` must be a unified currency code string for `this.currency (code)` to work on line 1815.

If `symbols` is an array of symbols and it contains exactly one symbol – take the `base` code from the associated  market to fill-in the `currency` in the request. We don't send the entire symbol, instead we just send the base part of it: https://github.com/ccxt/ccxt/wiki/Manual#consistency-of-base-and-quote-currencies

```JavaScript
        let code = undefined;
        if (symbols === undefined) {
            code = this.codeFromOptions ('fetchPositions', params);
        } else if (typeof symbols === 'string') {
            code = symbols;
        } else {
            if (Array.isArray (symbols)) {
                const length = symbols.length;
                if (length !== 1) {
                    throw new BadRequest (this.id + ' fetchPositions symbols argument cannot contain more than 1 symbol');
                }
                const market = this.market (symbols[0]);
                code = market['base'];
            }
        }
```